src/kernel/sum_slice/cuda.cu

```diff
@@ -26,7 +26,7 @@ void cuda_kernel(Index m, Index n, Index k, Index mk, T alpha, const T *src,
     Index i0 = threadIdx.x + blockIdx.x*blockDim.x,
           i1 = threadIdx.y + blockIdx.y*blockDim.y;
     Index i2_start = threadIdx.z, i2_step = blockDim.z;
-    constexpr T zero = 0;
+    constexpr scal_t zero = 0;
```

Comment:
This is incorrect, to be honest. But for now it is Ok. I mean in case of fp64_t or bf16_t this type shall not be scal_t.

Reply:
Actually, revert it back to type T